readchar/readchar_linux.py

```diff
@@ -1,18 +1,78 @@
 # -*- coding: utf-8 -*-
-# Initially taken from:
-# http://code.activestate.com/recipes/134892/
-# Thanks to Danny Yoo
+# Copyright (c) 2014 - 2017 Miguel Ángel García (@magmax9).
+# Licensed under the MIT license.
+# This code based on code snippet from the python 2 docs:
+# https://docs.python.org/2/faq/library.html
+#     #how-do-i-get-a-single-keypress-at-a-time
+
+
 import sys
-import tty
+import os
 import termios
+import fcntl
+from . import key
+
+
+def get_char(test_stream=None):
+    charbuffer = ''
+    readstream = read_char
+
+    if test_stream:
+        def altreader(*args):
+            return test_stream()
+        readstream = altreader
+
+    while True:
+        if charbuffer in key.ESCAPE_SEQUENCES:
+            char1 = readstream(False)
+        else:
+            char1 = readstream()
```

Comment:
Maybe it is my repo, but you contribute too much to not hearing you.

What about a private method getting if it is an escape sequence or not?:
```
char1 = readstream(not _is_escape(charbuffer))
```